spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonClientConfiguration.java

```diff
@@ -200,7 +200,11 @@ protected OverrideRestClient(IClientConfig ncc,
 		public URI reconstructURIWithServer(Server server, URI original) {
 			String scheme = original.getScheme();
 			if (!"https".equals(scheme) && this.serverIntrospector.isSecure(server)) {
-				original = UriComponentsBuilder.fromUri(original).scheme("https").build(true)
+				original = UriComponentsBuilder.fromUri(original)
+						.scheme("https")
+						.host(server.getHost())
+						.port(server.getPort())
+						.build(true)
 						.toUri();
 			}
 			return super.reconstructURIWithServer(server, original);
```

Comment:
Yeah, the problem was toUri() call tried to create an URI without the host and port and threw an invalid URI exception so it could never reach the super call.